comp/core/workloadmeta/collectors/internal/nvml/nvml.go

```diff
@@ -233,6 +233,17 @@ func (c *collector) Pull(_ context.Context) error {
 			gpu.SMCount = int(devAttr.MultiprocessorCount)
 		}
 
+		procs, ret := dev.GetComputeRunningProcesses()
```

Comment:
consider refactoring the Pull method to call multiple helper methods, as it became already long enough